docs/source/tutorial_optimization.rst

```diff
@@ -0,0 +1,891 @@
+
+One possible way for advanced RB methods is a reduction along the path
+of optimization. The idea is, that we start with an empty basis and only
+enrich the model with the parameters that we will arive at. This
+approach goes beyond the classical offline/online splitting of RB
+methods since it entirely skips the offline phase. In the following
+code, we will test this method.
+
+
+
+.. jupyter-execute::
+
+    pdeopt_reductor = CoerciveRBReductor(
+        fom, product=fom.energy_product, coercivity_estimator=coercivity_estimator)
+
+In the next function, we implement the above mentioned way of enriching
+the basis along the path of optimization.
+
+.. jupyter-execute::
+
+    def record_results_and_enrich(function, data, opt_dict, mu):
+        U = fom.solve(mu)
+        try:
+            pdeopt_reductor.extend_basis(U)
+            data['enrichments'] += 1
+        except:
+            print('Extension failed')
+        opt_rom = pdeopt_reductor.reduce()
+        QoI = rom.output(mu)
+        data['num_evals'] += 1
+        # we need to make sure to copy the data, since the added mu will be changed inplace by minimize afterwards
+        data['evaluation_points'].append([fom.parameters.parse(mu)['diffusion'][:][0], 
+                                          fom.parameters.parse(mu)['diffusion'][:][1]])
+        data['evaluations'].append(QoI[0])
+        opt_dict['opt_rom'] = rom 
+        return QoI
+    
+    def compute_gradient_with_opt_rom(opt_dict, mu):
+        opt_rom = opt_dict['opt_rom']
+        return opt_rom.output_d_mu(opt_rom.parameters.parse(mu), return_array=True, use_adjoint=True)
+
+.. jupyter-execute::
+
+    opt_along_path_minimization_data = {'num_evals': 0,
+                                           'evaluations' : [],
+                                           'evaluation_points': [],
+                                           'time': np.inf,
+                                           'enrichments': 0}
+    opt_dict = {} 
+    tic = perf_counter()
+    opt_along_path_result = minimize(partial(record_results_and_enrich, rom_objective_functional, 
+                                             opt_along_path_minimization_data, opt_dict),
+                                      initial_guess.to_numpy(),
+                                      method='L-BFGS-B', 
+                                      jac=partial(compute_gradient_with_opt_rom, opt_dict),
+                                      bounds=(ranges, ranges),
+                                      options={'ftol': 1e-15})
+    opt_along_path_minimization_data['time'] = perf_counter()-tic
+
+
+
+.. jupyter-execute::
+
+    report(opt_along_path_result, opt_along_path_minimization_data, reference_mu)
+
+    
+
+
+The computational time looks at least better than the FOM optimization
+and we are very close to the reference parameter saved some
+computational time. But we are following the exact same path than the
```

Comment:
I do not understand the last part of the sentence and I am not sure about what you want to actually say here.

Reply:
For every optimization there is one enrichment step which requires to solve the FOM. I added this. 
Does it now make sense to you?

Reply:
I think we are talking about different sentences, could that be? I do not understand the last part of 

> The computational time looks at least better than the FOM optimization and we are very close to the reference parameter saved some computational time.

Maybe it is just a language issue or I am just unable to understand what you mean here.

Reply:
Ah, yes wrong sentence. 
No you are absolutely right, the "saved some computational time" part is completely obsolete, thanks.